ast_decompiler/decompiler.py

```diff
@@ -425,17 +425,20 @@ def visit_withitem(self, node: ast.withitem) -> None:
             self.write(" as ")
             self.visit(node.optional_vars)
 
-    def visit_Try(self, node: ast.Try) -> None:
+    def visit_Try(self, node: Union[ast.Try, "ast.TryStar"]) -> None:
         self.write_indentation()
         self.write("try:")
         self.write_newline()
         self.write_suite(node.body)
+        is_trystar = sys.version_info >= (3, 11) and isinstance(node, ast.TryStar)
         for handler in node.handlers:
-            self.visit(handler)
+            self.visit_ExceptHandler(handler, is_trystar=is_trystar)
         self.write_else(node.orelse)
         if node.finalbody:
             self.write_finalbody(node.finalbody)
 
+    visit_TryStar = visit_Try
```

Comment:
You could also do this like this:

```python
    def visit_Try(self, node: Union[ast.Try, "ast.TryStar"], is_trystar: bool = False) -> None:
        self.write_indentation()
        self.write("try:")
        self.write_newline()
        self.write_suite(node.body)
        for handler in node.handlers:
            self.visit_ExceptHandler(handler, is_trystar=is_trystar)
        self.write_else(node.orelse)
        if node.finalbody:
            self.write_finalbody(node.finalbody)

    def visit_TryStar(self, node: "ast.TryStar") -> None:
        self.visit_Try(node, is_trystar=True)
```

But this is just personal taste rather than _accuracy_ per se :)

Reply:
Yes, and that's actually how I implemented AsyncWith, so maybe I should at least be consistent. I don't like though that that solution has redundant information in the two method arguments.